### ISSUE TITLE
Equivalent of Spark flatMap

### ISSUE THREAD

**Issue by crockpotveggies** [deeplearning4j/DataVec#365]
#### Issue Description

Currently we have lots of column ops, but nothing that allows me to perform row ops. Essentially, I need to duplicate records and obfuscate them in specific ways to increase available training data.

#### Version Information

master

#### Contributing

Happy to what I can here with some guidance.

**Comment by AlexDBlack** [comment]
This won't be trivial to add, as it will require some API changes.

As noted in Gitter, Transforms are a map operation - i.e., 1:1 op. This would add a 1:N op.

Two workarounds I can think of:
- Use Spark FlatMap operations on ```RDD<List<Writable>>``` outside of DataVec transforms
- convert to sequence of length 1 (convertToSequence()), expand the single value "sequence" into a longer "sequence" as required, convertFromSequence() to go back to independent records

**Comment by huitseeker** [comment]
`flatmap` is litterally a map + flatten operation. So, if you are able to write any `map : T => List<T>`, all we need is to add the flatten operation and a bit of syntactic sugar to hide those stages as a `flatmap`.

Spark, back when it cared about Dataframes as a special type had a function called explode, which was basically "flatten + caring about schemas" (now still existing but deprecated, as Datasets have made it easier to express this in terms of flatmap directly, embedding the schema into what's in the DataSet itself).

https://spark.apache.org/docs/2.1.0/api/java/org/apache/spark/sql/Dataset.html

That back-and-forth to sequence you're talking about, @AlexDBlack, is essentially what this represents — it does need a better frontend though.

**Comment by AlexDBlack** [comment]
Right, the *spark execution component* is trivial. The Transforms API will need some changes, which is the harder part (not major though).
Unless it's a critical feature (and the workarounds aren't suitable), I might hold off on this given the proposed changes to DataVec (see Basecamp).
